server/src/db/mod.rs

```diff
@@ -0,0 +1,106 @@
+pub mod models;
+pub mod schema;
+
+use anyhow::Result;
+use diesel::{pg::PgConnection, r2d2};
+use serde::Deserialize;
+
+pub type Pool = r2d2::Pool<r2d2::ConnectionManager<PgConnection>>;
+
+#[derive(Clone, Deserialize)]
+/// Configuration params for database. See https://docs.rs/postgres/0.17.3/postgres/config/struct.Config.html
+pub struct DatabaseSettings {
+    pub user: String,
+    pub password: String,
+    pub host: String,
+    pub port: u16,
+    pub dbname: String,
+    pub application_name: Option<String>,
+    pub connect_timeout_sec: u64,
+    pub pool_timeout_ms: u64,
+    pub read_timeout_ms: u64,
+}
+
+impl DatabaseSettings {
+    /// Obtain a [postgres url](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING)
+    /// with the given configuration parameters
+    fn to_postgres_url(&self) -> String {
+        let mut url = format!(
+            "postgres://{}:{}@{}:{}/{}",
```

Comment:
nit: move the constant format string at the top of the file?

Reply:
Seems a bit overkill, since it's only used here

Reply:
And if it were to change, the args to `format!` would also have to change, so they might as well be together